torch/onnx/symbolic_opset9.py

```diff
@@ -2194,3 +2194,14 @@ def _weight_norm(g, weight_v, weight_g, dim):
         return g.op("Mul", div, weight_g)
     else:
         return g.op("ATen", weight_v, weight_g, dim_i=dim, operator_s="_weight_norm")
+
+def dim(g, self):
+    '''Implement the dim functionality available for a pytorch tensor in ONNX'''
+    # For tracing we can directly use the shape of the tensor
+    rank = self.type().dim()
```

Comment:
thinking about it, maybe it would be better to always return the subgraph instead of self.type().dim().
This way we are guaranteed to have a dynamic value of dim.